packages/core/src/request.ts

```diff
@@ -161,6 +163,38 @@ export class Request {
         this.headers = { ...headers };
         this.userData = { ...userData };
         this.handledAt = handledAt as unknown instanceof Date ? (handledAt as Date).toISOString() : handledAt!;
+
+        Object.defineProperty(this.userData, '__crawlee', {
+            // eslint-disable-next-line no-underscore-dangle
+            value: this.userData.__crawlee,
+            enumerable: false,
+            writable: true,
+        });
+
+        Object.defineProperty(this.userData, 'toJSON', {
+            value: () => {
+                return {
+                    ...this.userData,
+                    toJSON: undefined as any,
```

Comment:
why?

Reply:
Brain fart from the past, implemented `toJSON` using `JSON.stringify()` (and tried to fix the recursion this way 🥴).